x/superfluid/keeper/concentrated_liquidity_test.go

```diff
@@ -0,0 +1,247 @@
+				return
+			}
+			suite.Require().NoError(err)
+
+			expectedNewCoins := sdk.NewCoins(sdk.NewCoin(clPool.GetToken0(), amount0.Add(tc.amount0Added)), sdk.NewCoin(clPool.GetToken1(), amount1.Add(tc.amount1Added)))
+
+			clPoolDenom := cltypes.GetConcentratedLockupDenomFromPoolId(clPool.GetId())
+			expectedLockCoins := sdk.NewCoins(sdk.NewCoin(clPoolDenom, newLiquidity.TruncateInt()))
+
+			// Define error tolerance
+			var errTolerance osmomath.ErrTolerance
+			errTolerance.AdditiveTolerance = sdk.NewDec(1)
+			errTolerance.RoundingDir = osmomath.RoundDown
+
+			// Resulting position should have the expected amount of coins within one unit (rounding down).
+			suite.Require().Equal(0, errTolerance.Compare(expectedNewCoins[0].Amount, finalAmount0), fmt.Sprintf("expected (%s), actual (%s)", expectedNewCoins[0].Amount, finalAmount0))
+			suite.Require().Equal(0, errTolerance.Compare(expectedNewCoins[1].Amount, finalAmount1), fmt.Sprintf("expected (%s), actual (%s)", expectedNewCoins[1].Amount, finalAmount1))
+
+			// Check the new lock.
+			newLock, err := suite.App.LockupKeeper.GetLockByID(ctx, newLockId)
+			suite.Require().NoError(err)
+			suite.Require().Equal(suite.App.StakingKeeper.UnbondingTime(ctx), newLock.Duration)
+			suite.Require().True(newLock.EndTime.IsZero())
+			suite.Require().Equal(poolJoinAcc.String(), newLock.Owner)
+			suite.Require().Equal(expectedLockCoins.String(), newLock.Coins.String())
+
+			// Check that a new position and lock ID were generated.
+			suite.Require().NotEqual(positionId, newPositionId)
+			suite.Require().NotEqual(lockId, newLockId)
+
+			// Check if intermediary account connection for the old lock ID is deleted.
+			oldIntermediaryAcc := superfluidKeeper.GetLockIdIntermediaryAccountConnection(ctx, lockId)
+			suite.Require().Equal(oldIntermediaryAcc.String(), "")
+
+			// Check if intermediary account connection for the new lock ID is created.
+			expAcc := types.NewSuperfluidIntermediaryAccount(clPoolDenom, valAddr.String(), 0)
+			newIntermediaryAcc := superfluidKeeper.GetLockIdIntermediaryAccountConnection(ctx, newLockId)
+			suite.Require().Equal(expAcc.GetAccAddress().String(), newIntermediaryAcc.String())
+
+			// Check if synthetic lockup for the old lock ID is deleted.
+			_, err = lockupKeeper.GetSyntheticLockup(ctx, lockId, keeper.StakingSyntheticDenom(clPoolDenom, valAddr.String()))
+			suite.Require().Error(err)
+
+			// Check if synthetic lockup for the new lock ID is created.
+			_, err = lockupKeeper.GetSyntheticLockup(ctx, newLockId, keeper.StakingSyntheticDenom(clPoolDenom, valAddr.String()))
+			suite.Require().NoError(err)
+
+			// Check if the old intermediary account has no delegation.
+			_, found := stakingKeeper.GetDelegation(ctx, oldIntermediaryAcc, valAddr)
+			suite.Require().False(found)
+
+			// Check if the new intermediary account has delegation.
+			_, found = stakingKeeper.GetDelegation(ctx, newIntermediaryAcc, valAddr)
+			suite.Require().True(found)
+
```

Comment:
Can we add check for staking denom supply check? (to ensure that appropriate amount of stake denom have been minted through this action)
Also can we add a check to check the increase of liquiidty in the pool?

Reply:
Added both denom supply check and pool address funds increase check https://github.com/osmosis-labs/osmosis/pull/5042/commits/31cbf91f44678a9c4a90dcc2073aa3f2a24dfc1c